pkg/actors/health/health.go

```diff
@@ -0,0 +1,297 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package health
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+
+	"k8s.io/utils/clock"
+
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	defaultInitialDelay           = time.Second * 1
+	defaultFailureThreshold       = int32(2)
+	defaultRequestTimeout         = time.Second * 2
+	defaultHealthyStateInterval   = time.Second * 3
+	defaultUnHealthyStateInterval = time.Second / 2
+	defaultSuccessStatusCode      = http.StatusOK
+)
+
+var healthLogger = logger.NewLogger("actorshealth")
+
+type options struct {
+	client                 *http.Client
+	address                string
+	initialDelay           time.Duration
+	requestTimeout         time.Duration
+	failureThreshold       int32
+	healthyStateInterval   time.Duration
+	unhealthyStateInterval time.Duration
+	successStatusCode      int
+	clock                  clock.WithTicker
+}
+
+// Option is a function that applies a health check option.
+type Option func(o *options)
+
+// Checker is a health checker which reports to whether the target actor app
+// endpoint is in a healthy state, according to the configured status code.
+type Checker struct {
+	client *http.Client
+
+	ch chan bool
+
+	address                string
+	initialDelay           time.Duration
+	requestTimeout         time.Duration
+	failureThreshold       int32
```

Comment:
Why not `int`? Or if it has to be 32-bit, why not `uint32`?

Reply:
Just because that was what was there before. I agree that we should be using `int`.